js_modules/dagit/packages/core/src/assets/AssetPartitions.tsx

```diff
@@ -207,73 +224,131 @@ export const AssetPartitions: React.FC<Props> = ({
+                      <MenuItem
+                        text={
+                          <Tooltip content="The order in which partitions were created, reversed">
+                            <Box flex={{direction: 'row', alignItems: 'center', gap: 8}}>
+                              <span>Reverse creation sort</span>
+                              <Icon name="info" />
+                            </Box>
+                          </Tooltip>
+                        }
+                        active={SortType.REVERSE_CREATION === sortType}
+                        onClick={() => {
+                          setSortTypes((sorts) => {
+                            const copy = [...sorts];
+                            copy[idx] = SortType.REVERSE_CREATION;
+                            return copy;
+                          });
+                        }}
+                        data-testId={testId('sort-reverse-creation')}
+                      />
+                      <MenuItem
+                        text="Alphabetical sort"
+                        active={SortType.ALPHABETICAL === sortType}
+                        onClick={() => {
+                          setSortTypes((sorts) => {
+                            const copy = [...sorts];
+                            copy[idx] = SortType.ALPHABETICAL;
+                            return copy;
+                          });
+                        }}
+                        data-testId={testId('sort-alphabetical')}
+                      />
+                      <MenuItem
+                        text="Reverse alphabetical sort"
+                        active={SortType.REVERSE_ALPHABETICAL === sortType}
+                        onClick={() => {
+                          setSortTypes((sorts) => {
+                            const copy = [...sorts];
+                            copy[idx] = SortType.REVERSE_ALPHABETICAL;
+                            return [...copy];
+                          });
+                        }}
+                        data-testId={testId('sort-reverse-alphabetical')}
+                      />
+                    </Menu>
                   }
-                  const dimensionKeyIdx = selection.dimension.partitionKeys.indexOf(dimensionKey);
-                  return partitionStatusAtIndex(
-                    rangesForEachDimension[idx]!,
-                    dimensionKeyIdx,
-                  ).filter((s) => statusFilters.includes(s));
-                }}
-                focusedDimensionKey={focusedDimensionKeys[idx]}
-                setFocusedDimensionKey={(dimensionKey) => {
-                  setFocusedDimensionKey(idx, dimensionKey);
-                }}
-              />
-            )}
-          </Box>
-        ))}
+                >
+                  <SortButton style={{marginRight: '-16px'}} data-testid={`sort-${idx}`}>
+                    <Icon name="sort_by_alpha" color={Colors.Gray400} />
+                  </SortButton>
```

Comment:
The positioning of the popover feels a bit off -- some right offset would be helpful here so that the menu doesn't sit on top of the trigger button.